.github/dump_issues.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+
+import csv
+import json
+import argparse
+import subprocess
+from pathlib import Path
+
```

Comment:
Need to improve it from error handling point of view:

```
kiparis@kiparis-pc:~/projects/scylla-cluster-tests$ python .github/dump_issues.py --repository "scylladb/scylla-bench"
Traceback (most recent call last):
  File "/home/kiparis/projects/scylla-cluster-tests/.github/dump_issues.py", line 85, in <module>
    main()
  File "/home/kiparis/projects/scylla-cluster-tests/.github/dump_issues.py", line 81, in main
    dump_issue_status_to_csv(args.repository, args.target_dir)
  File "/home/kiparis/projects/scylla-cluster-tests/.github/dump_issues.py", line 57, in dump_issue_status_to_csv
    assert all_issues, f'No issues found for repository {repository}'
AssertionError: No issues found for repository scylladb/scylla-bench
kiparis@kiparis-pc:~/projects/scylla-cluster-tests$ python .github/dump_issues.py --repository "scylladb/scylla-bench"
> /home/kiparis/projects/scylla-cluster-tests/.github/dump_issues.py(53)dump_issue_status_to_csv()
-> ret = subprocess.run(q, shell=True, text=True, capture_output=True, check=True)
(Pdb) n
> /home/kiparis/projects/scylla-cluster-tests/.github/dump_issues.py(54)dump_issue_status_to_csv()
-> pages = json.loads(ret.stdout)
(Pdb) n
> /home/kiparis/projects/scylla-cluster-tests/.github/dump_issues.py(56)dump_issue_status_to_csv()
-> all_issues = sum([p['data']['search']['nodes'] for p in pages], [])
(Pdb) n
> /home/kiparis/projects/scylla-cluster-tests/.github/dump_issues.py(58)dump_issue_status_to_csv()
-> assert all_issues, f'No issues found for repository {repository}'
(Pdb) all_issues
[]
(Pdb) pages
[]
(Pdb) ret
CompletedProcess(args='\n    gh api graphql --paginate -f query=\'\n    query Search($endCursor: String){\n      search(query: "repo:scylladb/scylla-bench", type: ISSUE, first: 100,  after: $endCursor) {\n        issueCount\n        pageInfo {\n          startCursor\n          hasNextPage\n          endCursor\n        }\n        nodes {\n          ... on PullRequest {\n            number\n            title\n            state\n            labels(first: 50, orderBy: {direction: DESC, field: CREATED_AT}) {\n              nodes {\n                name\n              }\n            }\n          }\n          ... on Issue {\n            number\n            title\n            state\n            labels(first: 50, orderBy: {direction: DESC, field: CREATED_AT}) {\n              nodes {\n                name\n              }\n            }\n          }\n        }\n        pageInfo {\n          endCursor\n          hasNextPage\n        }\n      }\n    }\' | jq -s .\n    ', returncode=0, stdout='[]\n', stderr='/bin/sh: 2: gh: not found\n')

```

Reply:
> Need to improve it from error handling point of view:
> 
> ```
> kiparis@kiparis-pc:~/projects/scylla-cluster-tests$ python .github/dump_issues.py --repository "scylladb/scylla-bench"
> Traceback (most recent call last):
>   File "/home/kiparis/projects/scylla-cluster-tests/.github/dump_issues.py", line 85, in <module>
>     main()
>   File "/home/kiparis/projects/scylla-cluster-tests/.github/dump_issues.py", line 81, in main
>     dump_issue_status_to_csv(args.repository, args.target_dir)
>   File "/home/kiparis/projects/scylla-cluster-tests/.github/dump_issues.py", line 57, in dump_issue_status_to_csv
>     assert all_issues, f'No issues found for repository {repository}'
> AssertionError: No issues found for repository scylladb/scylla-bench
> 
> 
> kiparis@kiparis-pc:~/projects/scylla-cluster-tests$ python .github/dump_issues.py --repository "scylladb/scylla-bench"
> > /home/kiparis/projects/scylla-cluster-tests/.github/dump_issues.py(53)dump_issue_status_to_csv()
> -> ret = subprocess.run(q, shell=True, text=True, capture_output=True, check=True)
> (Pdb) n
> > /home/kiparis/projects/scylla-cluster-tests/.github/dump_issues.py(54)dump_issue_status_to_csv()
> -> pages = json.loads(ret.stdout)
> (Pdb) n
> > /home/kiparis/projects/scylla-cluster-tests/.github/dump_issues.py(56)dump_issue_status_to_csv()
> -> all_issues = sum([p['data']['search']['nodes'] for p in pages], [])
> (Pdb) n
> > /home/kiparis/projects/scylla-cluster-tests/.github/dump_issues.py(58)dump_issue_status_to_csv()
> -> assert all_issues, f'No issues found for repository {repository}'
> (Pdb) all_issues
> []
> (Pdb) pages
> []
> (Pdb) ret
> CompletedProcess(args='\n    gh api graphql --paginate -f query=\'\n    query Search($endCursor: String){\n      search(query: "repo:scylladb/scylla-bench", type: ISSUE, first: 100,  after: $endCursor) {\n        issueCount\n        pageInfo {\n          startCursor\n          hasNextPage\n          endCursor\n        }\n        nodes {\n          ... on PullRequest {\n            number\n            title\n            state\n            labels(first: 50, orderBy: {direction: DESC, field: CREATED_AT}) {\n              nodes {\n                name\n              }\n            }\n          }\n          ... on Issue {\n            number\n            title\n            state\n            labels(first: 50, orderBy: {direction: DESC, field: CREATED_AT}) {\n              nodes {\n                name\n              }\n            }\n          }\n        }\n        pageInfo {\n          endCursor\n          hasNextPage\n        }\n      }\n    }\' | jq -s .\n    ', returncode=0, stdout='[]\n', stderr='/bin/sh: 2: gh: not found\n')
> 
> ```

It's intended to be run by GitHub action, 

And it's a bit of a shame you don't have a GitHub cli installed :)

Also you couldn't run the code in the action even before this change...

Reply:
> It's intended to be run by GitHub action,

As of now, if it fails for any reason, you won't know it not rerunning it.
Not critical, but useful change.

> And it's a bit of a shame you don't have a GitHub cli installed :)

I don't use GH API, only web UI. So, it is nonsense to install something just "because".

Reply:
change it to print error, so now you can install it :)

you were using the script for doing backport before ? 

also how do you checkout PR that of backport ? I never remember the git syntax for it

while this I can remember: 
```
gh pr checkout 7850
```

Reply:
> changed it to print error, so now you can install it :)

Good

> you were using the script for doing backport before ?

We have 2 utils scripts, only 1 of them uses the `gh`.

> also how do you checkout PR that of backport ? I never remember the git syntax for it
> 
> while this I can remember:
> 
> ```
> gh pr checkout 7850
> ```

I used only `git`:
- `git remote add ...` (once)
- `git fetch ...` (per-pr)
- `git checkout ... ` (per-pr)